### DISCUSSION TITLE
Type issue when following docs example to embed a form in Livewire

### DISCUSSION THREAD

**Discussion by adamhopkinson** [filamentphp/filament#14418] [Help] [answered]
### Package

Form builder

### Package Version

v3.2.111

### How can we help you?

I'm following the example in the [docs](https://filamentphp.com/docs/3.x/forms/adding-a-form-to-a-livewire-component) almost verbatim to add a form to a livewire page. My code looks like this:

```php
?php

namespace App\Livewire;

use Filament\Actions\Action;
use Filament\Forms\Components\TextInput;
use Filament\Forms\Concerns\InteractsWithForms;
use Filament\Forms\Contracts\HasForms;
use Illuminate\Support\Facades\Log;
use Livewire\Component;

class SubscriptionSignupPage extends Component implements HasForms
{
    use InteractsWithForms;

    public ?array $data = [];

    public function mount(): void
    {
        $this->form->fill();
    }

    public function form(\Livewire\Form $form): \Livewire\Form
    {
        return $form
            ->schema([
                TextInput::make('email')
                    ->required(),
            ])
            ->statePath('data');
    }
```

When I visit this page, I get an Internal Server Error:

```
App\Livewire\SubscriptionSignupPage::form(): Argument #1 ($form) must be of type Livewire\Form, Filament\Forms\Form given
```

This points to the declaration of the `form` method (hence why I've been explicit about the types of the `$form` attribute and return value).

I can easily avoid this by not setting the attribute and return types (ie leaving them undefined or `mixed`) or by setting them to `Filament\Forms\Form`, but I'm curious as to why this is happening and whether the docs should be updated or if it's something I'm doing wrong elsewhere (eg because I'm using a full page livewire component).

**Comment by dissto** [comment, marked as answer]
Mhh the link to the docs you gave do say to use `use Filament\Forms\Form;`, or what do you mean?

![image](https://github.com/user-attachments/assets/56451ac5-4556-4c68-a39a-5ddd75ca14b8)

**Reply by adamhopkinson** [reply]
Oh wow, I think I've been staring at this too long! Thanks for your help @dissto
